pulsar-broker/src/test/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannelTest.java

```diff
@@ -0,0 +1,772 @@
+                throw new RuntimeException(e);
+            }
+        });
+        errorCnt = validateChannelStart(channel);
+        closeFuture.get();
+        assertTrue(errorCnt > 0);
+        errorCnt = validateChannelStart(channel);
+        assertEquals(6, errorCnt);
+    }
+
+    private int validateChannelStart(ServiceUnitStateChannelImpl channel) {
+        int errorCnt = 0;
+        try {
+            channel.isChannelOwner();
+        } catch (IllegalStateException e) {
+            errorCnt++;
+        }
+        try {
+            channel.getChannelOwner();
+        } catch (IllegalStateException e) {
+            errorCnt++;
+        }
+        try {
+            channel.getOwnerAsync(bundle);
+        } catch (IllegalStateException e) {
+            errorCnt++;
+        }
+        try {
+            channel.publishAssignEventAsync(bundle, lookupServiceAddress1);
+        } catch (IllegalStateException e) {
+            errorCnt++;
+        }
+        try {
+            channel.publishUnloadEventAsync(
+                    new Unload(lookupServiceAddress1, bundle, Optional.of(lookupServiceAddress2)));
+        } catch (IllegalStateException e) {
+            errorCnt++;
+        }
+        try {
+            channel.publishSplitEventAsync(new Split(bundle, lookupServiceAddress1, Map.of()));
+        } catch (IllegalStateException e) {
+            errorCnt++;
+        }
+        return errorCnt;
+    }
+
+    @Test(priority = 1)
+    public void compactionScheduleTest() throws PulsarServerException {
+
+        channel1.scheduleCompaction();
+
+        Awaitility.await()
+                .pollInterval(200, TimeUnit.MILLISECONDS)
+                .atMost(5, TimeUnit.SECONDS)
+                .untilAsserted(() -> { // wait until true
+                    try {
+                        var threshold = admin.topicPolicies()
+                                .getCompactionThreshold(ServiceUnitStateChannelImpl.TOPIC, false).longValue();
+                        assertEquals(5 * 1024 * 1024, threshold);
+                    } catch (Exception e) {
+                        ;
+                    }
+
+                });
+    }
+
+    @Test(priority = 2)
+    public void assignmentTest()
+            throws ExecutionException, InterruptedException, IllegalAccessException, TimeoutException {
+        var getOwnerRequests1 = spy(getOwnerRequests((channel1)));
+        var getOwnerRequests2 = spy(getOwnerRequests((channel2)));
+
+        var owner1 = channel1.getOwnerAsync(bundle);
+        var owner2 = channel2.getOwnerAsync(bundle);
+
+        assertNull(owner1);
+        assertNull(owner2);
+
+        var assigned1 = channel1.publishAssignEventAsync(bundle, lookupServiceAddress1);
+        var assigned2 = channel2.publishAssignEventAsync(bundle, lookupServiceAddress2);
+        assertNotNull(assigned1);
+        assertNotNull(assigned2);
+        waitUntilOwnerChanges(channel2, bundle, null);
```

Comment:
```suggestion
        waitUntilOwnerChanges(channel1, bundle, null);
```

Reply:
I think adding the `Command` layer is additional complexity and requires more lines of code. Also, the current approach does not need to store the additional `Command` info in the system topic(space optimized).

Regarding `ServiceUnitState.isValidTransition()`, the state validation(aka compaction) will happen in the tableview and topic compaction by `TopicCompactionStrategy`(which is in a separate PIP and PR here, https://github.com/apache/pulsar/issues/18099). This validation logic, ServiceUnitChannelCompactionStrategy will be added after this PR https://github.com/apache/pulsar/pull/18195 gets merged.

Reply:
Thanks for the `waitUntilOwnerChanges(channel1, bundle, null);` catch.

updated.

Reply:
> I think adding the `Command` layer is additional complexity and requires more lines of code. Also, the current approach does not need to store the additional `Command` info in the system topic(space optimized).
> 
> Regarding `ServiceUnitState.isValidTransition()`, the state validation(aka compaction) will happen in the tableview and topic compaction by `TopicCompactionStrategy`(which is in a separate PIP and PR here, #18099). This validation logic, ServiceUnitChannelCompactionStrategy will be added after this PR #18195 gets merged.

Thanks for the explanation, I'm not quite sure if my proposal is reasonable. I'll wait for other features to be added and revisit